### PR TITLE
[RFC] vim-patch:7.4.2065

### PR DESCRIPTION
**Pull request by Shougo**
Problem:    Compiler warns for uninitialzed variable. (John Marriott)
Solution:   Set lnum to the right value.

https://github.com/vim/vim/commit/69aa099641616268bfcde9dc3aa313c677846b12

### CODE REVIEW

**Comment by Shougo** [comment]
I will fix the compilation errors.

**Comment by Shougo** [comment]
Fixed.

File: src/nvim/eval.c [hunk 1]
```diff
@@ -9573,46 +9573,49 @@ static void f_foldlevel(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
     }
 
     /* Find interesting text in this line. */
     s = skipwhite(ml_get(lnum));
     /* skip C comment-start */
     if (s[0] == '/' && (s[1] == '*' || s[1] == '/')) {
       s = skipwhite(s + 2);
-      if (*skipwhite(s) == NUL
-          && lnum + 1 < (linenr_T)vimvars[VV_FOLDEND].vv_nr) {
+      if (*skipwhite(s) == NUL && lnum + 1 < foldend) {
```
Comment (justinmk) [inline src/nvim/eval.c]: Where did this change come from? It's not upstream.

Comment (justinmk) [inline src/nvim/eval.c]: I see, it makes sense.

**Comment by justinmk** [comment]
LGTM. 

Also incorporate sneaky changes from:                                                                                        
vim-patch:7.4.2063 73dad1e64cb42842d8259cb1a255a6fa59822f76

**Comment by justinmk** [comment]
Test failed:

```
Executing Test_wrong_arguments()
Executed 20 tests
1 FAILED
Caught exception: Vim(function):E126: Missing :endfunction @ /home/travis/build/neovim/neovim/src/nvim/testdir/test_expr.vim, line 108


From test_alot.vim:
Caught exception: Vim(function):E126: Missing :endfunction @ /home/travis/build/neovim/neovim/src/nvim/testdir/test_expr.vim, line 108

Test results:


From test_alot.vim:
Caught exception: Vim(function):E126: Missing :endfunction @ /home/travis/build/neovim/neovim/src/nvim/testdir/test_expr.vim, line 108
TEST FAILURE
```

**Comment by Shougo** [comment]
Thank you

**Pull request merged**